tests/unit/sidecars/live_data_watcher_tests.py

```diff
@@ -53,6 +93,23 @@ def test_s3_load_type_on_change(self):
         self.assertEqual(dest.owner(), pwd.getpwuid(os.getuid()).pw_name)
         self.assertEqual(dest.group(), grp.getgrgid(os.getgid()).gr_name)
 
+    def test_s3_load_type_on_change_sharding(self):
+        dest = self.output_dir.joinpath("data.txt")
+        inst = NodeWatcher(str(dest), os.getuid(), os.getgid(), 777)
+
+        new_content = b'{"live_data_watcher_load_type":"S3","bucket_name":"test_bucket","file_key":"test_file_key","sse_key":"test_decryption_key","region_name":"us-east-1", "num_file_shards": 5}'
+        expected_content = b'{"foo_encrypted": "bar_encrypted"}'
+
+        # For safe measure, run this 50 times. It should succeed every time.
+        # We've uploaded 5 files to S3 in setUp() and num_file_shards=5 in the
```

Comment:
Out of curiosity, is there a way to enrich this test such that we verify that every one of the five prefixes was read from at least once?

Reply:
Not sure if I have any ideas since random number generation doesn't guarantee that we hit all the 5 prefixes even if we ran it 1000 times (very unlikely, but certainly possible). It would probably make the unit test flakey (failing a very small percentage of the time).

There might be a different way to do what you're saying though...but let me investigate.

Reply:
Yeah, flakiness is a good worry.

If it's possible to check which prefixes get read, maybe we'd be happy if it's at least two unique ones?

Reply:
@areitz That sounds reasonable! Should I still only run over it 50 times or should I increase that?

Reply:
Oooh thanks for the pointer Roman - let me see if I can figure this out quickly - if not, and if this becomes more urgent, I may have to merge without that unit test.